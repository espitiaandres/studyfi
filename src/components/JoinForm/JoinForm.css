
.heading {
    border-bottom: 4px groove white;
    color: white;
    padding-bottom: 10px;
}
  
.duplicatemessage {
    color: white;
}
  
.button {
    background: #1ED760;
    border: none;
    border-radius: 5px;
    display: inline-block;
    padding: 20px;
    text-decoration: none;
    width: 300px;
}

.seasonStyling {
    background: #F37D0F;
}

.seasonStylingAltHeader {
    border-bottom: 4px groove #FFE01B;
}

.button:hover {
    cursor: pointer;
    filter: brightness(70%);
}

button:focus {
    outline: 0;
}
  
.mt-20 {
    margin-top: 20px;
}
  
@media (min-width: 320px) and (max-width: 480px) {
    .joinOuterContainer {
      height: 100%;
    }
  
    .joinInnerContainer {
      width: 90%;
    }
}
  