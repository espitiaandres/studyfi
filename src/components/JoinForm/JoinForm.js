//
//  JoinForm.js
//  react-spotify-player
//
//  Created by Andres Espitia.
//  Copyright Â© 2020 Andres Espitia. All rights reserved.
//

import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import './JoinForm.css';

const JoinForm = ({ season }) => {    
    const [name, setName] = useState('');
    const [room, setRoom] = useState('');
    const [tz, setTZ] = useState('');

    let seasonStyling = season ? "seasonStyling" : "";
    let seasonStylingAlt = season ? "seasonStylingAlt" : "";

    const onJoinFormSubmit = (e) => {
        e.preventDefault();
    }

    let timezones = [];

    for (let i = 12; i >= 1; i--) {
        timezones.push("GMTplus" + i.toString());
    }

    timezones.push("GMT");

    for (let i = 1; i <= 11; i++) {
        timezones.push("GMTminus" + i.toString());
    }

    return (
        <form onSubmit={onJoinFormSubmit}>
            <h1 className={`heading ${seasonStylingAlt}Header`}>chat</h1>
            <div>
                <input placeholder="name" className="joinInput" type="text" onChange={(event) => setName(event.target.value)} />
            </div>
            <div>
                <input placeholder="room" className="joinInput mt-20" type="text" onChange={(event) => setRoom(event.target.value)} />
            </div>
            <div>
                <select className="joinInput mt-20" defaultValue={'default'} onChange={(event) => setTZ(event.target.value)}>
                    <option value='default'>choose your timezone</option>
                    {timezones.map(
                        (tzs) => <option value={tzs}>
                                    {tzs.includes("plus") ? tzs.replace("plus", "+") : tzs.replace("minus", "-")}
                                </option>
                        )
                    }
                </select>
            </div>
            <Link 
                onClick={e => (!name || !room || !tz) ?  e.preventDefault() : null} 
                to={`/chat?name=${name}&room=${room}&tz=${tz}`}
            >
                <button className={`button mt-20 ${seasonStyling}`} type="submit">sign in with google</button>
            </Link>
        </form>
    )
}

export default JoinForm
