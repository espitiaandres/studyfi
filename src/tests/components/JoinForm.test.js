//
//  JoinForm.test.js
//  react-spotify-player
//
//  Created by Andres Espitia.
//  Copyright Â© 2020 Andres Espitia. All rights reserved.
//

import React from 'react';
import { shallow } from 'enzyme';
import JoinForm from '../../components/JoinForm/JoinForm';

test('should render JoinForm component correctly', () => {
    const wrapper = shallow(<JoinForm />);
    expect(wrapper).toMatchSnapshot();
});
