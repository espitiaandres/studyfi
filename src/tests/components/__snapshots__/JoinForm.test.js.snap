// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render JoinForm component correctly 1`] = `ShallowWrapper {}`;
